src/battle-scene.ts

```diff
@@ -2506,32 +2507,48 @@ export default class BattleScene extends SceneBase {
    * @returns
    */
   getMysteryEncounter(override: MysteryEncounter): MysteryEncounter {
+    const biomeMysteryEncounters = mysteryEncountersByBiome.get(this.arena.biomeType);
     // Loading override or session encounter
     let encounter: MysteryEncounter;
-    if (!Utils.isNullOrUndefined(Overrides.MYSTERY_ENCOUNTER_OVERRIDE) && Overrides.MYSTERY_ENCOUNTER_OVERRIDE < allMysteryEncounters.length) {
-      encounter = allMysteryEncounters[Overrides.MYSTERY_ENCOUNTER_OVERRIDE];
+    /*
+    if (!Utils.isNullOrUndefined(Overrides.MYSTERY_ENCOUNTER_OVERRIDE) && Overrides.MYSTERY_ENCOUNTER_OVERRIDE < biomeMysteryEncounters.length) {
+      encounter = biomeMysteryEncounters[Overrides.MYSTERY_ENCOUNTER_OVERRIDE];
     } else {
-      encounter = override?.encounterType >= 0 ? allMysteryEncounters[override?.encounterType] : null;
+      encounter = override?.encounterType >= 0 ? biomeMysteryEncounters[override?.encounterType] : null;
     }
-
+    */
+    console.log(encounter);
     // Generate new encounter if no overrides
     if (!encounter) {
       const tierValue = Utils.randSeedInt(64);
       let tier = tierValue > 32 ? MysteryEncounterTier.COMMON : tierValue > 16 ? MysteryEncounterTier.UNCOMMON : tierValue > 6 ? MysteryEncounterTier.RARE : MysteryEncounterTier.SUPER_RARE;
       let availableEncounters = [];
-
+      for (encounter of biomeMysteryEncounters) {
+        console.log("-------" + MysteryEncounterType[encounter.encounterType] + " Encounter Check -------");
+        console.log(encounter);
+        console.log( "Encountercheck: " + encounter.meetsRequirements(this));
+        console.log( "protagCheck: " +  encounter.meetsProtagonistRequirementAndProtagonistPokemonSelected(this));
+        console.log( "supportCheck: " +  encounter.meetsSupportingRequirementAndSupportingPokemonSelected(this));
+        console.log(MysteryEncounterTier[encounter.encounterTier]);
+      }
       // If no valid encounters exist at tier, checks next tier down, continuing until there are some encounters available
       while (availableEncounters.length === 0 && tier >= 0) {
-        availableEncounters = allMysteryEncounters.filter((encounter) => encounter?.meetsRequirements(this) && encounter.encounterTier === tier);
+
+        availableEncounters = biomeMysteryEncounters.filter((encounter) => encounter?.meetsRequirements(this) &&
+        encounter.meetsSupportingRequirementAndSupportingPokemonSelected(this) && // support is checked first to handle cases of protagonist overlapping with support
```

Comment:
Before an encounter triggers, we check additional requirements in both supporting and protagonist pokemon